spritecontrol.cpp

```diff
@@ -14,24 +14,31 @@ SpriteGenerator::SpriteGenerator() {
 
 	PaletteGroup palette_group = (PaletteGroup) (cfg[Cfg::Palette]);
 
-	std::vector<const PaletteData *> bag_of_palettes;
-	if (palette_group == PaletteGroup::RandomlyGenerated) {
+	std::vector<Palettes::Definition> bag_of_palettes;
+	if (cfg[Cfg::UseCustomPalettes] == 1.0f) {
+		bag_of_palettes = PaletteRepository().get_all_custom_palettes();
+	} else if (palette_group == PaletteGroup::RandomlyGenerated) {
 		for (int i = Cfg::MaxColors.range.first; i < Cfg::MaxColors.range.second; i++) {
 			bag_of_palettes.push_back(RandomPalettes::random(i));
 		}
 	} else {
 		auto group = PaletteGroups::get(palette_group);
-		for (const auto &palette : group.members) {
-			bag_of_palettes.push_back(palette.data);
-		}
+		bag_of_palettes = group.members;
+	}
+
+	if (bag_of_palettes.empty()) {
+		bag_of_palettes = PaletteGroups::All.members;
 	}
 
+	int min_colors = bag_of_palettes.size() < Cfg::MaxColors.range.first
+		? bag_of_palettes.size()
+		: Cfg::MaxColors.range.first;
```

Comment:
Normally you can't have less than 2 palettes, but I wanted to let people run with just a single palette if they really wanted to, AND to have a reasonable default if someone just defined no palettes at all. These extra checks to `bag_of_palettes.size()` protect against that.